src/curve/constant-product/ConstantProductPair.sol

```diff
@@ -118,18 +120,18 @@ contract ConstantProductPair is ReservoirPair {
     }
 
     function mint(address aTo) external override returns (uint256 rLiquidity) {
-        (uint104 lReserve0, uint104 lReserve1, uint32 lBlockTimestampLast,) = _lockAndLoad();
-        (lReserve0, lReserve1) = _syncManaged(lReserve0, lReserve1); // check asset-manager pnl
+        (uint256 lReserve0, uint256 lReserve1, uint32 lBlockTimestampLast,) = _lockAndLoad();
+        (lReserve0, lReserve1) = _syncManaged(uint104(lReserve0), uint104(lReserve1)); // check asset-manager pnl
```

Comment:
The point of using uint256 is to avoid the cost of casting, we lose this saving if we immediately cast. _syncManaged should accept uint256, the only place we should perform (safe) casting is in `_lockAndLoad()` and `updateAndUnlock()`